### ISSUE TITLE
ErrorException: unserialize(): Error at offset 3726497 of 3727350 bytes

### ISSUE THREAD

**Issue by sheldonkotyk** [statamic/v2-hub#1212]
### Expected behaviour
Save should save an entry

### Actual behaviour
After multiple entries are worked on (edited usually), saving an entry causes an exception

### Server configuration
**Operating system**:
ubuntu
**Web server:**
nginx
**PHP version:**
7.0
**Statamic version:**
2.5.1

**Updated from an older Statamic or fresh install:**
updated from 2.1.19 -> 2.5 -> 2.5.1

[statamic-2017-01-30.log.zip](https://github.com/statamic/v2-hub/files/740962/statamic-2017-01-30.log.zip)

**Comment by jackmcdade** [comment]
The times i've seen that were usually when there was some malformed data in the YAML somewhere, which created an issue with the cache. Does this happen every time, or just sometimes?

**Comment by sheldonkotyk** [comment]
They say its pretty consistently after the 5th article they work on. First 4 work fine. May be race condition related.

**Comment by sheldonkotyk** [comment]
clearing the cache gives them another 5 articles or so.

**Comment by warkior** [comment]
We're also seeing this error on images at times.
/img/asset/MzgwZGM4...VyLmpwZw==

Error
----------------
ErrorException: unserialize(): Error at offset 32730 of 32758 bytes

Location
----------------
/home/forge/thesitedomain.com/releases/20170303001950/statamic/vendor/laravel/framework/src/Illuminate/Cache/FileStore.php:78 - Illuminate\Foundation\Bootstrap\HandleExceptions::handleError

**Comment by jackmcdade** [comment]
By chance are you working with images that are either super skinny or super tall? Like 800x40 or 20x1000 or anything remotely like that?

**Comment by sheldonkotyk** [comment]
All different sizes.

https://issuesiface.com
https://thelife.com

**Comment by jackmcdade** [comment]
Right, but do you have _at least one_ of that really dramatic/odd size in there?

**Comment by sheldonkotyk** [comment]
I don't think we have anything that dramatic but I'll ask.

**Comment by wuhhh** [comment]
Same problem here with error at different line. This happens when saving a new entry - happening fairly consistently. Refresh gives me the "Entry Saved" message and the entry appears in the list with no further issues (everything is correctly saved). 

```
ErrorException in FileStore.php line 78:
unserialize(): Error at offset 8180 of 8182 bytes
```

**Comment by jackwakefield** [comment]
This is a race condition with `Illuminate\Cache\FileStore` where one request will attempt to unserialize a file while another is concurrently updating the file but hasn't yet finished. It was fixed in laravel/framework#12796.

**Comment by jackwakefield** [comment]
I'm running into this upwards of 30-50 times a day as my site is making several AJAX requests on load, it's making me really nervous for when this site enters production. I appreciate Statamic uses Laravel 5.2 and is most likely an unopened can of worms to update. Perhaps a Laravel 5.2 fork with backports applied is a (short term) option?

**Comment by jasonvarga** [comment]
Thanks for pointing out that PR. We'll see what we can do!

**Comment by jasonvarga** [comment]
Also, @jackwakefield you can probably set `stache_always_update: false` in your `site/settings/caching.yaml`

**Comment by jasonvarga** [comment]
I believe I have this fixed for 2.6.2. Let me know if it continues to happen for you.

**Comment by Neophen** [comment]
Still happening at version:  2.11.2
here's the code that produces this issues sometimes:

```php
    public function postPage()
    {
        $page = $this->page;
        $unset_fields = ["template", "fieldset", "slug", "url", "uri", "url_path", "permalink", "published", "order_type", "collection", "is_entry", "is_page", "edit_url"];

        collect(\Statamic\API\Config::getLocales())->map(function ($locale) use ($page, $unset_fields) {
            $page_in = $page->dataForLocale($locale);

            $draft_exists = false;
            if (array_key_exists("draft_exists", $page_in)) {
                $draft_exists = $this->convertToBool($page_in["draft_exists"]);
            }

            if (!$draft_exists) {
                // Log::debug("NO DRAFT for: " . $this->convertLocaleToLanguage($locale));
                $page->in($locale)->set("is_outdated", "true," . $locale)->save();
                $this->setPageDraftDeleted($page, $locale);
                return;
            } 

            // Log::debug("Draft exists for: " . $this->convertLocaleToLanguage($locale));
            $draft_id = $page->in($locale)->get('draft_id');
            if (!Entry::exists($draft_id)) {
                // protecting against ghost data.
                $this->setPageDraftDeleted($page, $locale);
                return;
            }

            $full_draft_data = Entry::find($draft_id)->dataForLocale($locale);

            $cleaned_data = $this->unsetMetaData($full_draft_data, $unset_fields);

            $page_update_data = array_merge($page_in, $cleaned_data);
            // Log::debug($page_update_data);
            $page->in($locale)->data($page_update_data)->save();
            //we set is_outdated after setting the data!
            $page->in($locale)->set("is_outdated", "false," . $locale)->save();
            $this->setPageDraftDeleted($page, $locale);
        });
    }
```

Here is the error message, should I open a new issue?
```
[2019-01-01 14:19:07] dev.ERROR: ErrorException: unserialize(): Error at offset 2 of 16374 bytes in /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Cache/FileStore.php:78
Stack trace:
#0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(8, 'unserialize(): ...', '/Users/mykolas/...', 78, Array)
#1 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Cache/FileStore.php(78): unserialize('s:21268:"{"page...')
#2 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Cache/FileStore.php(47): Illuminate\Cache\FileStore->getPayload('stache::timesta...')
#3 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Cache/Repository.php(98): Illuminate\Cache\FileStore->get('stache::timesta...')
#4 [internal function]: Illuminate\Cache\Repository->get('stache::timesta...', Array)
#5 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(318): call_user_func_array(Array, Array)
#6 /Users/mykolas/Sites/lcc/statamic/core/API/Cache.php(42): Illuminate\Cache\CacheManager->__call('get', Array)
#7 /Users/mykolas/Sites/lcc/statamic/core/Stache/UpdateManager.php(132): Statamic\API\Cache::get('stache::timesta...', Array)
(edited)
#8 /Users/mykolas/Sites/lcc/statamic/core/Stache/UpdateManager.php(117): Statamic\Stache\UpdateManager->getTimestamps()
#9 /Users/mykolas/Sites/lcc/statamic/core/Stache/UpdateManager.php(53): Statamic\Stache\UpdateManager->createTraverser(Object(Statamic\Stache\Drivers\TaxonomiesDriver))
#10 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Support/Collection.php(135): Statamic\Stache\UpdateManager->Statamic\Stache\{closure}(Object(Statamic\Stache\Drivers\TaxonomiesDriver), 'taxonomies')
#11 /Users/mykolas/Sites/lcc/statamic/core/Stache/UpdateManager.php(68): Illuminate\Support\Collection->each(Object(Closure))
#12 /Users/mykolas/Sites/lcc/statamic/core/Stache/Manager.php(80): Statamic\Stache\UpdateManager->update()
#13 /Users/mykolas/Sites/lcc/statamic/core/Providers/StacheServiceProvider.php(173): Statamic\Stache\Manager->update()
#14 [internal function]: Statamic\Providers\StacheServiceProvider->boot(Object(Illuminate\Http\Request))
#15 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Container/Container.php(507): call_user_func_array(Array, Array)
#16 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(734): Illuminate\Container\Container->call(Array)
#17 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(717): Illuminate\Foundation\Application->bootProvider(Object(Statamic\Providers\StacheServiceProvider))
#18 [internal function]: Illuminate\Foundation\Application->Illuminate\Foundation\{closure}(Object(Statamic\Providers\StacheServiceProvider), 30)
#19 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(718): array_walk(Array, Object(Closure))
#20 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/BootProviders.php(17): Illuminate\Foundation\Application->boot()
#21 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(203): Illuminate\Foundation\Bootstrap\BootProviders->bootstrap(Object(Statamic\Application))
#22 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(222): Illuminate\Foundation\Application->bootstrapWith(Array)
#23 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(117): Illuminate\Foundation\Http\Kernel->bootstrap()
#24 /Users/mykolas/Sites/lcc/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(87): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#25 /Users/mykolas/Sites/lcc/index.php(155): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#26 /Users/mykolas/.composer/vendor/laravel/valet/server.php(151): require('/Users/mykolas/...')
#27 {main}  
```
